lib/SILOptimizer/Transforms/SILSROA.cpp

```diff
@@ -249,9 +249,14 @@ void SROAMemoryUseAnalyzer::chopUpAlloca(std::vector<AllocStackInst *> &Worklist
   for (auto *LI : Loads) {
     SILBuilderWithScope B(LI);
     llvm::SmallVector<SILValue, 4> Elements;
-    for (auto *NewAI : NewAllocations)
-      Elements.push_back(B.createLoad(LI->getLoc(), NewAI,
-                                      LoadOwnershipQualifier::Unqualified));
+    for (auto *NewAI : NewAllocations) {
+      LoadOwnershipQualifier loadQual = LoadOwnershipQualifier::Unqualified;
+      if (LI->getFunction()->hasOwnership())
+        loadQual = NewAI->getType().isTrivial(*LI->getFunction())
```

Comment:
This is not needed. You should be using emitLoadValueOperation. That should work in all cases.